### ISSUE TITLE
No animation when returning the data array to default values

### ISSUE THREAD

**Issue by vladislav-dovzhenko-finsight** [clauderic/dnd-kit#1149]
I use SortableContext to animate the sorting of the table. The animation works fine, but when I return the data array to default values, the animation disappears.

https://codesandbox.io/s/serene-heisenberg-hmhttw?file=/src/App.js:1739-1754

Is there any solution to this problem?
Thanks

**Comment by ehrro** [comment]
The items Id needs to be a string.

**Comment by dontsave** [comment]
@ehrro strings do not solve the problem. see:

https://github.com/clauderic/dnd-kit/assets/3527177/7c4d0703-53d4-4663-a8aa-849a18a8416f

notice that resetting the order to the original (default values) does not trigger an animation even though the IDs are strings. this is a true bug in the library. the weird thing is i swear this was working at one point. perhaps it's a bug introduced by React 18? cc @clauderic have you encountered this?

**Comment by ehrro** [comment]
Try to use arrayMove function to set the state to default values.

**Comment by dontsave** [comment]
@ehrro i think you might be missing the bigger picture here. the `arrayMove` utility simply manipulates the array in the same way any other utility works. using it, as for example to swap the first two items:


https://github.com/clauderic/dnd-kit/assets/3527177/ccdf6127-b2c3-4eca-8398-6fff7423cb3e


doesn't solve the problem. this is a bug in the library

**Comment by wilq32-unused** [comment]
I agree with @ehrro, in example he provides, you can simply change sorting order and animation works there. There is something weird happening about default state of the order 🤔

**Comment by midoushitongtong** [comment]
try use **framer-motion** to solve the issue.


```jsx
import { motion } from "framer-motion";

const SortableItem = ({ item }) => (
  <motion.div layout>
    {item.name}
  </motion.div>
);
```

**Comment by clauderic** [comment]
This issue was opened against a previous version of dnd-kit. The library has since undergone a major rewrite.

It has been rebuilt from the ground up with support for **multiple frameworks** (React, Vue, Svelte, Solid, and vanilla JavaScript), improved performance, and a new API.

- **Documentation**: https://dndkit.com
- **Migration guide**: https://dndkit.com/react/migration
- **GitHub**: https://github.com/clauderic/dnd-kit

If this issue is still relevant to the latest version, please open a new issue.
